### ISSUE TITLE
Timestamps are being recorded as normalized to UTC by Iceberg Connector

### ISSUE THREAD

**Issue by tdcmeehan** [prestodb/presto#23361]
[According to the Iceberg spec](https://iceberg.apache.org/spec/#parquet), the `timestamp` type in Iceberg should map to the `TIMESTAMP_MICROS` logical type in Parquet.  `TIMESTAMP_MICROS` is parameterized and takes in two parameters: the unit (here, it is microseconds), and whether or not it is adjusted to UTC.  According to the Iceberg spec, the UTC adjustment must be disabled.  This is because [according to the Parquet spec](https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#local-semantics-timestamps-not-normalized-to-utc), timestamps without UTC adjustment represent a local time (i.e., a time that is independent of user's timezone).

In Presto, we are always writing `timestamp` values with UTC adjustment enabled:

```
tdcmeehan@Timothys-MBP ~ % parquet meta /var/folders/q5/h4rx7vn15jxdw3ysr50qsw5w0000gn/T/PrestoTest2735905965489955697/iceberg_data/HIVE/tpch/test_timestamp/data/e9f9fe92-09b6-41dd-8721-448d89e9aa27.parquet 

File path:  /var/folders/q5/h4rx7vn15jxdw3ysr50qsw5w0000gn/T/PrestoTest2735905965489955697/iceberg_data/HIVE/tpch/test_timestamp/data/e9f9fe92-09b6-41dd-8721-448d89e9aa27.parquet
Created by: null
Properties: (none)
Schema:
message table {
  optional int64 x (TIMESTAMP(MICROS,true)) = 1;
}


Row group 0:  count: 1  56.00 B records  start: 4  total(compressed): 56 B total(uncompressed):56 B 
--------------------------------------------------------------------------------
   type      encodings count     avg size   nulls   min / max
x  INT64     G RRD     1         56.00 B    0       "2017-05-01T10:12:34.00000..." / "2017-05-01T10:12:34.00000..."
```

The `true` value in the column `x` `TIMESTAMP(MICROS,true)` means it is normalized to UTC.

This is true regardless if the flag `deprecated.legacy-timestamp` is set to true or false.  This is incorrect, because the Iceberg type represents a timestamp without a time zone, and hence it should not be normalized to any particular time zone.

**Comment by hantangwangd** [comment]
As I understand, should we record the type of column x in parquet metadata as `TIMESTAMP(MICROS, false)`, no matter `deprecated.legacy-timestamp` is set to true or false? As the concept of `legacy-timestamp` is inside presto engine.

**Comment by tdcmeehan** [comment]
@hantangwangd yes, I think we can just universally specify that this is not normalized to UTC, as this doesn't change regardless of the value of the flag.

**Comment by tdcmeehan** [comment]
Closed by #23388
